packages/fast-components-react-msft/src/pivot/pivot.tsx

```diff
@@ -0,0 +1,240 @@
+     * Returns tabs managedclasses with new carousel-specific JSS
+     */
+    protected generatePivotClassNames(): TabsClassNameContract {
+        return {
+            tabs: get(this.props, "managedClasses.pivot", ""),
+            tabs_tabPanels: this.generateTabPanelsClassNames(),
+            tabs_tabList: get(this.props, "managedClasses.pivot_tabList", ""),
+            tabs_tabContent: get(this.props, "managedClasses.pivot_tabContent", ""),
+            tabs_tabPanelContent: get(
+                this.props,
+                "managedClasses.pivot_tabPanelContent",
+                ""
+            ),
+            tab: get(this.props, "managedClasses.pivot_tab", ""),
+            tab__active: get(this.props, "managedClasses.pivot_tab__active", ""),
+            tabPanel: get(this.props, "managedClasses.pivot_tabPanel", ""),
+            tabPanel__hidden: get(
+                this.props,
+                "managedClasses.pivot_tabPanel__hidden",
+                ""
+            ),
+        };
+    }
+
+    private handleTabsUpdate = (activeTabId: string): void => {
+        this.setState({
+            activeId: activeTabId,
+        });
+    };
+
+    private isSelected(element: HTMLElement): boolean {
+        return element.className.includes("active") === true;
+    }
+
+    private updateTabPanelIndex(): void {
+        if (canUseDOM() && this.tabsRef.current && Array.isArray(this.props.items)) {
+            const tabElement: HTMLElement = ReactDOM.findDOMNode(
+                this.tabsRef.current
+            ) as HTMLElement;
+
+            const mytabsArray: HTMLElement[] = Array.prototype.slice.call(
+                tabElement.querySelectorAll("[role='tab']")
+            );
+            this.setState({
+                tabPanelIndex: mytabsArray.findIndex(this.isSelected),
+            });
+        }
+    }
+
+    private generateTabPanelsClassNames(): string {
+        let className: string = get(this.props, "managedClasses.pivot_tabPanels", "");
+        if (this.state.tabPanelIndex === this.prevTabPanelIndex) {
+            className = className;
+        } else if (this.state.tabPanelIndex < this.prevTabPanelIndex) {
+            className = `${className} ${get(
+                this.props,
+                "managedClasses.pivot_tabPanels__previous",
+                ""
+            )}`;
+        } else {
+            className = `${className} ${get(
+                this.props,
+                "managedClasses.pivot_tabPanels__next",
+                ""
+            )}`;
+        }
+        this.prevTabPanelIndex = this.state.tabPanelIndex;
+        return className;
+    }
+
+    private setActiveIndicatorOffset(): void {
+        if (canUseDOM() && this.tabsRef.current && Array.isArray(this.props.items)) {
+            const tabElement: HTMLElement = ReactDOM.findDOMNode(
+                this.tabsRef.current
+            ) as HTMLElement;
+
+            const mytabs: NodeListOf<Element> = tabElement.querySelectorAll(

```

Comment:
It looks like you only ever want a single element here - if so, you should use `tabElement.querySelector` instead of `tabElement.querySelectorAll` - `querySelector` will return the first matching element.